_tutorials/faceted-search.md

```diff
@@ -0,0 +1,1845 @@
+      "store_location": {
+        "type": "geo_point"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Index some documents containing different stores into the index:
+
+```json
+POST /stores/_bulk
+{ "index": { "_id": "1" } }
+{ "name": "Downtown Store", "store_location": { "lat": 40.7510, "lon": -73.9900 } }
+{ "index": { "_id": "2" } }
+{ "name": "Suburban Store", "store_location": { "lat": 40.8300, "lon": -74.2000 } }
+{ "index": { "_id": "3" } }
+{ "name": "Outskirts Store", "store_location": { "lat": 41.2000, "lon": -74.8000 } }
+```
+{% include copy-curl.html %}
+
+For location-based faceting (like finding products available at nearby stores), use the `geo_distance` aggregation:
+
+```json
+POST /stores/_search
+{
+  "aggs": {
+    "store_distance": {
+      "geo_distance": {
+        "field": "store_location",
+        "origin": "40.7507, -73.9895",
+        "unit": "mi",
+        "ranges": [
+          { "to": 5, "key": "Within 5 miles" },
+          { "from": 5, "to": 25, "key": "5-25 miles away" },
+          { "from": 25, "key": "Over 25 miles away" }
+        ]
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The response contains all three stores bucketed by distance:
+
+<details markdown="block">
+  <summary>
+    Response
+  </summary>
+  {: .text-delta}
+
+```json
+{
+  "took": 16,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 3,
+      "relation": "eq"
+    },
+    "max_score": 1,
+    "hits": [
+      {
+        "_index": "stores",
+        "_id": "1",
+        "_score": 1,
+        "_source": {
+          "name": "Downtown Store",
+          "store_location": {
+            "lat": 40.751,
+            "lon": -73.99
+          }
+        }
+      },
+      {
+        "_index": "stores",
+        "_id": "2",
+        "_score": 1,
+        "_source": {
+          "name": "Suburban Store",
+          "store_location": {
+            "lat": 40.83,
+            "lon": -74.2
+          }
+        }
+      },
+      {
+        "_index": "stores",
+        "_id": "3",
+        "_score": 1,
+        "_source": {
+          "name": "Outskirts Store",
+          "store_location": {
+            "lat": 41.2,
+            "lon": -74.8
+          }
+        }
+      }
+    ]
+  },
+  "aggregations": {
+    "store_distance": {
+      "buckets": [
+        {
+          "key": "Within 5 miles",
+          "from": 0,
+          "to": 5,
+          "doc_count": 1
+        },
+        {
+          "key": "5-25 miles away",
+          "from": 5,
+          "to": 25,
+          "doc_count": 1
+        },
+        {
+          "key": "Over 25 miles away",
+          "from": 25,
+          "doc_count": 1
+        }
+      ]
+    }
+  }
+}
+```
+
+</details>
+
+## Advanced faceting scenarios
+
+The following sections provide examples of advanced faceted search functionality.
+
+### Post-filter faceting
```

Comment:
I feel this can be moved to section `Filter by facet values`. 
Also I don't think we need to mention the keyword `post-filter`, but just describe the function or scenario.
Like the user selects a facet, the query results are filtered for that facet, but the facet selector is kept unchanged.

I think another thing worth mention is global aggregation https://docs.opensearch.org/latest/aggregations/bucket/global/. It always executed on the full dataset, seems a good way to have a unchanged facet selector.